processor/src/main/java/org/jboss/pnc/processor/ClientGenerator.java

```diff
@@ -204,10 +143,107 @@ private boolean process0(Set<? extends TypeElement> annotations, RoundEnvironmen
         return true;
     }
 
-    private void addDefaultParameters(ExecutableElement restApiMethod, MethodSpec.Builder builder) {
+    private MethodSpec buildMethod(ExecutableElement restApiMethod, boolean withParameters) {
+        String parametersList = getParameters(restApiMethod);
+
+        TypeMirror returnType = restApiMethod.getReturnType();
+        TypeMirror returnGeneric = null; //TODO check for null before adding a generic (currently no such case)
+        if (returnType instanceof DeclaredType) {
+            if (((DeclaredType)returnType).getTypeArguments().size() > 0) {
+                returnGeneric = ((DeclaredType)returnType).getTypeArguments().get(0);
+            }
+        }
+
+        MethodSpec.Builder builder = MethodSpec.methodBuilder(restApiMethod.getSimpleName().toString())
+                .addModifiers(Modifier.PUBLIC);
+
+        builder.addException(ClassName.get("org.jboss.pnc.client", "RemoteResourceException"))
+                .beginControlFlow("try");
+
+        if (withParameters) {
+            builder.addStatement("PncClientRequestParams.setParams(parameters)");
```

Comment:
Any particular reason for a filter and a ThreadLocal variable ?

Reply:
It's ugly but I couldn't find a better way to pass the query params.
I'm open to suggestions.